modules/cache.js

```diff
@@ -0,0 +1,171 @@
+var async = require('async');
+var transactionTypes = require('../helpers/transactionTypes.js');
+var client, self, logger, cacheEnabled, cacheReady = true, errorCacheDisabled = 'Cache Unavailable';
+
+// Constructor
+function Cache (cb, scope) {
+	self = this;
+	client = scope.cache.client;
+	logger = scope.logger;
+	cacheEnabled = scope.cache.cacheEnabled;
+	setImmediate(cb, null, self);
+}
+
+Cache.prototype.isConnected = function () {
+	return cacheEnabled && client.connected;
+};
+
+Cache.prototype.isReady = function () {
+	return cacheReady && self.isConnected();
+};
+
+Cache.prototype.getJsonForKey = function (key, cb) {
+	// we can use config var to check if caching is activated
+	if (self.isConnected()) {
+		client.get(key, function (err, value) {
+			// parsing string to json
+			cb(err, JSON.parse(value));
+		});
+	} else {
+		cb(errorCacheDisabled);
+	}
+};
+
+Cache.prototype.setJsonForKey = function (key, value, cb) {
+	if (self.isConnected()) {
+		// redis calls toString on objects, which converts it to object [object] so calling stringify before saving
+		client.set(key, JSON.stringify(value), cb);
+	} else {
+		cb(errorCacheDisabled);
+	}
+};
+
+Cache.prototype.deleteJsonForKey = function (key, cb) {
+	if (self.isConnected()) {
+		client.del(key, cb);
+	} else {
+		cb(errorCacheDisabled);
+	}
+};
+
+Cache.prototype.removeByPattern = function (pattern, cb) {
+	if (self.isConnected()) {
```

Comment:
Can we skip one nesting level globally in that file?
`if (!self.isConnected()) { return cb(errorCacheDisabled); }`